portfolio/src/main/webapp/script.js

```diff
@@ -39,24 +40,157 @@ function refreshComments() {
   }
 }
 
+/** Obtain comment data from datastore and display them on page */
 function getData() {
   refreshComments();
-  var numContactsToDisplay = document.getElementById('quantity').value;
-  var link = '/data?numContactsToDisplay=' + numContactsToDisplay
+  const numContactsToDisplay = document.getElementById('quantity').value;
+  let link = '/data?numContactsToDisplay=' + numContactsToDisplay;
 
+  // fetch  the data and create a list containing the message of each contact
   fetch(link).then(response => response.json()).then((contacts) => {
+    console.log('Hello World');
+    const infoContainer = document.getElementById('more-info');
     contacts.forEach((contact) => {
-      var currContact = document.createElement('li');
-      var userInfo = document.createTextNode(
+      const currContact = document.createElement('li');
+      const userInfo = document.createTextNode(
           contact.firstName + ' ' + contact.lastName + ' says ' +
           contact.subject);
       currContact.appendChild(userInfo);
-      document.getElementById('more-info').appendChild(currContact);
+      infoContainer.appendChild(currContact);
     });
   });
 }
 
 function deleteData() {
+  // upon request navigate to the delete servlet and remove all comments from
+  // datastore
   const request = new Request('/delete-data', {method: 'POST'});
   fetch(request).then((results) => getData());
 }
+
+/** Allow the user to login, when logged in -> display contact form */
+function userLogin() {
+  fetch('/login').then((response) => response.json()).then((currUser) => {
+    document.getElementById('form-overlay-text').innerHTML =
+        currUser.loginMessage;
+    if (currUser.isLoggedIn) {
+      document.getElementById('form-overlay').style.display = 'block';
+    } else {
+      document.getElementById('form-overlay').style.display = 'none';
+    }
+    // display delete all comments button to one specified user => admin
+    if (currUser.email == 'eretana@google.com') {
+      document.getElementById('delete-btn').style.display = 'inline';
+    } else {
+      document.getElementById('delete-btn').style.display = 'none';
+    }
+  });
+}
+
+/** Initialize a map utlizing google's Map API for web */
+function initMap() {
+  const markerPos = {lat: 31.770581604323954, lng: -106.50421142578125};
+  const marker =
+      new google.maps.Marker({position: markerPos, title: 'UTEP! My school.'});
+  const map = new google.maps.Map(document.getElementById('map'), {
+    center: {lat: 31.7869021, lng: -106.3127764},
+    zoom: 15,
+    // set styles for dark mode
+    styles: [
+      {elementType: 'geometry', stylers: [{color: '#242f3e'}]},
```

Comment:
create constants for these colors and use descriptive names for the colors: 
https://google.github.io/styleguide/jsguide.html#naming-constant-names

ie. #242f3e can be DARK_BLUE